exporter/opentelemetry-exporter-zipkin/tests/test_zipkin_exporter.py

```diff
@@ -289,11 +286,21 @@ def test_export(self):
             status = exporter.export(otel_spans)
             self.assertEqual(SpanExportResult.SUCCESS, status)
 
-        mock_post.assert_called_with(
-            url="http://localhost:9411/api/v2/spans",
-            data=json.dumps(expected),
```

Comment:
Would reversing this (loading the data into a dict instead of dumping `expected`) work without writing the comparison code?

```py
kwargs = mock_post.call_args[1]
kwargs['data'] = json.loads(kwargs['data'])
self.assertEqual(kwargs, {"url": "http://localhost:9411/api/v2/spans", "data": expected, ...})
```

Reply:
I don't think so. The culprit is a JSON object that is serialized as a string so if we parse it, the JSON doc will be parsed a string field and can have nondeterministic ordering. I think this is good enough to fix broken master for now. It is a bit more code but the code is not cryptic IMO and is clear to understand even if a bit more verbose which is better for tests anyway :)

Reply:
You mean parts of the JSON are double encoded? If not Python doesn't seem to care about dict ordering for deep equality.

Reply:
Yeah, pre-python 3.6 dicts were not ordered as far as i remember.